### ISSUE TITLE
Pruning is broken

### ISSUE THREAD

**Issue by Neonkraft** [automl/ConfigurableOptimizer#103]
[This line](https://github.com/automl/ConfigurableOptimizer/blob/2015987c7e22e7aee3224164fc2ba0c77aaa184c/src/confopt/searchspace/darts/core/model_search.py#L501) and the subsequent are incorrect.

1. `p.data` does not have gradients by default. It's just the data of the tensor without grad info.
2. You can't set requires_grad at a cell level in a tensor. Either the whole tensor requires grad, or it doesn't.
3. `p.data *= mask.float()` sets the pruned alphas to 0. The problem is that the alpha values may be negative values (they have to be positive only after the softmax operation), so by setting the lowest negative value to 0, you're actually increasing the rank of that operation in the subsequent epochs. E.g, let's say the original alpha values were [-5, -1, 1, 5]. After pruning the weakest operation it becomes [0, -1, 1, 5]. This means that in the next epoch, it actually becomes stronger than the second weakest operation (which has a strength of -1).
